woocommerce/woocommerce
# dev: migrate @woocommerce/components/link to ts

### All Submissions:


### Changes proposed in this Pull Request:

- Renamed source, test and storybook files for the Link component to .tsx
- Added type annotations where necessary
- Added handling for undefined type errors in Link component, particularly around event.target.closest

Closes #35376.

### How to test the changes in this Pull Request:



1. Run `pnpm --filter=@woocommerce/components run test` - all unit tests should pass
1. Run `pnpm --filter=@woocommerce/storybook run storybook`
1. Go to http://localhost:6007/
1. Test `Link`
1. Smoke test around the components that Link is used in

![image](https://user-images.githubusercontent.com/27843274/210552352-73fa74c0-e39b-476a-a90d-2921a218ac24.png)



### Other information:




### FOR PR REVIEWER ONLY:
